backend/src/common/services/calculation-progress/calculation-progress.service.ts

```diff
@@ -62,6 +73,25 @@ class CalculationProgressService {
                             : null,
                 },
             );
+
+            if (
+                calculateGoalProgress(goal, filteredWorkouts) ===
+                    COMPLETED_GOAL_VALUE &&
+                updatedGoal
+            ) {
+                hasCompletedGoal = true;
+            }
+        }
+
+        if (hasCompletedGoal) {
+            await this.notificationService.create({
+                userId,
+                title: 'Goal Completed',
+                message:
+                    'Congratulations! You have completed your goal. Go to Goals Page to see your results.',
```

Comment:
Maybe it's better to save messages in enum. What do you think?

Reply:
Fixed)